File: Python/8kyu/8kyu-are-you-playing-banjo.py
# Are You Playing Banjo?
# Language: Python
# Version: 3.10
# Difficulty: 8 kyu

def are_you_playing_banjo(name):
    return name + " plays banjo" if name[0].upper() == "R" else name + " does not play banjo"


# Tests
import codewars_test as test

try:
    from solution import areYouPlayingBanjo as are_you_playing_banjo
except ImportError:
    from solution import are_you_playing_banjo
   
@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(are_you_playing_banjo("martin"), "martin does not play banjo");
        test.assert_equals(are_you_playing_banjo("Rikke"), "Rikke plays banjo");
        test.assert_equals(are_you_playing_banjo("bravo"), "bravo does not play banjo")
        test.assert_equals(are_you_playing_banjo("rolf"), "rolf plays banjo")

File: R/8kyu/8kyu-are-you-playing-banjo.r
# Are You Playing Banjo?
# Language: R
# Version: 3.4.1
# Difficulty: 8 kyu

# Solution 1
are_you_playing_banjo <- function(name){
  name_split <- unlist(strsplit(name, NULL))
  ifelse(toupper(name_split[1]) == "R", paste(name, "plays banjo"), paste(name, "does not play banjo"))
}

# Solution 2 using grepl() 
are_you_playing_banjo <- function(name){
  ifelse(grepl("^(R|r)", name), paste(name, "plays banjo"), paste(name, "does not play banjo"))
}

# Another grepl variation useful when more variables
grepl("^(S|T|R|I|N|G)", name, ignore.case = TRUE)


# Tests
test_that('Basic tests', {
  expect_equal(are_you_playing_banjo("Adam"), "Adam does not play banjo")
  expect_equal(are_you_playing_banjo("Paul"), "Paul does not play banjo")
  expect_equal(are_you_playing_banjo("Ringo"), "Ringo plays banjo");
  expect_equal(are_you_playing_banjo("bravo"), "bravo does not play banjo")
  expect_equal(are_you_playing_banjo("rolf"), "rolf plays banjo")
})